Source/Master/src/main.cpp

```diff
@@ -13,62 +13,86 @@
 #define SERVER_ADDRESS "SERVER IP"
 #define SERVER_PORT 80
 
+#define I2C_
+#define WIFI
+
 WiFiHandler wifiHandler = WiFiHandler();
 WebSocketHandler webSocketHandler = WebSocketHandler();
-I2C i2c;
+I2C i2c (LOCAL_ADDRESS);
+
+#define SCAN_OFFSET 200 * 1000 // 10 Seconds
+int scanTime = 0;
 
 void setup()
 {
-    // WiFiData wifiData(SSID, PASSWORD);
-    // wifiHandler = WiFiHandler(wifiData);
-
-    // WebSocketData webSocketData(SERVER_ADDRESS, SERVER_PORT);
-    // webSocketHandler = WebSocketHandler(webSocketData);
-
-    // wifiHandler.Connect();
-    // if(!wifiHandler.isConnected())
-    // {
-    //   return;
-    // }
-
-    // webSocketHandler.Connect();
-    // if(!webSocketHandler.isConnected())
-    // {
-    //   return;
-    // }
+    #ifdef WIFI
+
+    WiFiData wifiData(SSID, PASSWORD);
+    wifiHandler = WiFiHandler(wifiData);
+
+    WebSocketData webSocketData(SERVER_ADDRESS, SERVER_PORT);
+    webSocketHandler = WebSocketHandler(webSocketData);
+
+    wifiHandler.Connect();
+    if (!wifiHandler.isConnected())
+    {
+        return;
+    }
+
+    webSocketHandler.Connect();
+    if (!webSocketHandler.isConnected())
+    {
+        return;
+    }
+    #endif
     Serial.begin(9600);
     Serial.println();
     Serial.println("Starting up!");
-    i2c = I2C(LOCAL_ADDRESS);
+    #ifdef I2C_
+    scanTime = millis() + SCAN_OFFSET;
+    i2c.InitializeAddresses();
+    #endif
 }
 
-Node_Registers lastRequest = NR_None;
+int lastRequest;
+
 
 void loop()
 {
-    // if(webSocketHandler.isConnected())
-    // {
-    //   std::string websocketData = webSocketHandler.Recieve();
-    //   if(websocketData != "")
-    //   {
-    //     // Decompile with JSON parser
-    //     // Do some stuff with the data
+    #ifdef WIFI
+    if (webSocketHandler.isConnected())
+    {
+        std::string websocketData = webSocketHandler.Recieve();
+        if (websocketData != "")
+        {
+        // Decompile with JSON parser
```

Comment:
I assume these will come later? or is this logic elsewhere?

Reply:
This is for another branch.